### PR TITLE
Feature/harmen make testable

### PR DESCRIPTION
**Pull request by harmenjanssen**
This is the start of the test suite for this package.
It currently includes two test files for basic functions. Next step is to add fixed images and actually test whether sharp resizes them correctly (or more accurately: whether we give sharp the correct configuration to produce our desired output).

Changes in this PR:

- Adding Jest + two test cases.
- Adding a Github Actions workflow for running prettier, eslint and tests.
- Ignoring yarn.lock: the idea behind this is that we don't have to include the binary in this package. This way someone can run the tests on their own Mac, while still being able to deploy this from Github Actions and generate a Linux binary. What do you think of this? Obviously if you want to deploy this from your local machine you will still have to run `yarn install` using the directive that's specified in the README, because you will need to generate the Linux binary regardless of your own operating system.

Hopefully this paves the way for a more robust package. From here we can solve any edge-cases that come up and add these to the test suite.

### CODE REVIEW

**Comment by harmenjanssen** [comment]
A disadvantage of removing yarn.lock from the repository is the inability to cache the yarn dependencies.

File: test/pathToParams.test.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      expect(outputFormat).toBe("jpeg");
+      expect(originalExtension).toBe("jpg");
+    });
+  });
+  describe("Given a URL with just a width", () => {
+    it("Will still extract the dimensions", () => {
+      const input = "/500x/foo.jpg";
+      const [size, path, outputFormat, originalExtension] = pathToParams(
+        input,
+        ALLOWED_EXTENSIONS
```
Comment (martijngastkemper) [inline test/pathToParams.test.js:23]: I would replace it with a simple array to prevent confusion. To me it looks like it has something to do with the tests, but it doesn't. And it decouples the test, although that's a very small issue.

```suggestion
        ['jpg']
```

File: test/pathToParams.test.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
+const pathToParams = require("../util/pathToParams.js");
+
+describe("pathToParams", () => {
```
Comment (martijngastkemper) [inline test/pathToParams.test.js:3]: Maybe add a test case to verify allowed extensions?

**Review by martijngastkemper — COMMENTED** [review]
> Ignoring yarn.lock: the idea behind this is that we don't have to include the binary in this package. This way someone can run the tests on their own Mac, while still being able to deploy this from Github Actions and generate a Linux binary. What do you think of this? Obviously if you want to deploy this from your local machine you will still have to run yarn install using the directive that's specified in the README, because you will need to generate the Linux binary regardless of your own operating system.

Is this really true? `yarn install` puts the right binaries in place. The problem occurs when deploying from Mac to Linux. But when deploying from a workflow using Linux is not a problem. No MacOS involved.

So I think you can restore the `.lock` file and add a note to the readme that says to deploy you need the same platform as the Lambda. And add an example workflow file.

File: test/pathToParams.test.js (hunk 1 above)
Comment (harmenjanssen) [reply test/pathToParams.test.js:23]: Fair enough! Will do.

File: test/pathToParams.test.js (hunk 2 above)
Comment (harmenjanssen) [reply test/pathToParams.test.js:3]: Check. Will do!

**Comment by harmenjanssen** [comment]
> Is this really true? yarn install puts the right binaries in place. The problem occurs when deploying from Mac to Linux. But when deploying from a workflow using Linux is not a problem. No MacOS involved.
> 
> So I think you can restore the .lock file and add a note to the readme that says to deploy you need the same platform as the Lambda. And add an example workflow file.

Of course, I was thinking about this the wrong way! The binaries aren't in the repo at all, nor in `yarn.lock`.
I will return it to the repo, thanks for your feedback.

**Comment by harmenjanssen** [comment]
See last 4 commits. 
Let me know if you give the green light on this one. 🙂

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "12.x"
+      - run: yarn install
+      - name: Upload logs
+        run: npx eslint .
```
Comment (martijngastkemper) [inline .github/workflows/ci.yml:20]: `name` heeft denk ik de verkeerde waarde.

**Review by martijngastkemper — APPROVED** [review]
Één dingetje over een step name. Heel fijn dat dit er nu in zit. Geeft een hoop vertrouwen.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (harmenjanssen) [reply .github/workflows/ci.yml:20]: Yes, inderdaad.
